data/etl/movieshowing.py

```diff
@@ -12,38 +18,122 @@ class MovieShowing:
 
     def __init__(self, cinema):
         self.driver = webdriver.PhantomJS()
+        self.driver.set_window_size(1124, 850)  # set browser size
         self.cinema_id, self.cinema_name, self.cinema_url = cinema
 
     def extract_cinema_schedule(self):
         """retrieve one cinema schedule based on the given url,
-        return a list of dictionaries contains """
-        # retrieve title, (type like 3D) and schedule time
-        print(self.cinema_url)
+        return a list of dictionaries of movie titles
+
+        Main logic flow as follow:
+            1. extract_raw
+            2. transform and packaging
+            3. match and store imdb id
+            4. return the data
+        """
+        # self.extract_raw_golden_village()
+        # self.extract_raw_cathay()
+        self.extract_raw_shaw_brother()
+
+    # extract_raw
+    def extract_raw_golden_village(self):
         self.driver.get(self.cinema_url)
+        # retrieve title, (type like 3D) and schedule time raw data
+        tabs = self.driver.find_elements_by_class_name("ng-binding")
+        for tab in tabs:
+            if tab.get_attribute("ng-bind-html") == "day.day":
+                print(tab.text)
+                if tab.text == "Advance Sales":
+                    break
+
+                tab.click()
+                rows = self.driver.find_elements_by_class_name("row")
+                for row in rows:
+                    # get movie title
+                    anchors = row.find_elements_by_class_name("ng-binding")
+                    for anchor in anchors:
+                        if anchor.get_attribute("ng-bind-html") == "getFilmTitle(movie)":
+                            print(anchor.text)
 
-        # find imdb id
-        # create tuple cinema_id, movie_id, type, schedule
+                    # get movie schedule
+                    buttons = row.find_elements_by_css_selector("button")
+                    for button in buttons:
+                        if button.get_attribute("ng-bind-html") == "time.time":
+                            print(button.text)
+
+                            # find imdb id
+                            # create tuple cinema_id, movie_id, type, schedule
+        return
+
```

Comment:
maybe extract as separate functions for single responsibility? Nested loops make it hard to read, same for cathay extractor